### PR TITLE
[Session][VarDumper] Deprecate accepting legacy mongo extension

### PR DESCRIPTION
**Pull request by Tobion**
| Q             | A
| ------------- | ---
| Branch?       | 3.4 
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        | 

The old ext-mongo is deprecated and does not work with PHP 7. See http://php.net/manual/en/mongo.setup.php
People should use mongodb/mongodb and ext-mongodb instead. This deprecates functionality using the old ext-mongo classes.

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Tobion.

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php [hunk 1]
```diff
@@ -12,7 +12,12 @@
 namespace Symfony\Component\HttpFoundation\Session\Storage\Handler;
 
 /**
+ * Session handler using the mongodb/mongodb package and MongoDB driver extension.
+ *
  * @author Markus Bachmann <markus.bachmann@bachi.biz>
+ *
+ * @see https://packagist.org/packages/mongodb/mongodb
+ * @see http://php.net/manual/en/set.mongodb.php
```
Comment (stof) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php]: should use HTTPS

**Comment by nicolas-grekas** [comment]
PR welcomed on master to remove the legacy code.

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php (hunk 1 above)
Comment (Tobion) [reply src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php]: this should be changed everwhere then. we use http php.net documentation mostly

**Comment by Tobion** [comment]
On it.

**Pull request merged**